Draft: Feature/2 aws cdk

cchet opened the pull request:
Migrated the Terraform AWS infra to AWS CDK Java with container.
Please review the infrastructure and give it a try with any training available.
For instance with this bootstrap file https://github.com/Gepardec/containerization-training/tree/main/aws/bootstrap.sh 

Please read aws-cdk/REAMDE.adoc for the instructions.
This is still a draft and I would extend it with a github-action building the jar and the container image.

Feedback is appreciated.

Commit:
STarted AWS CDK Java infra definition

Commit:
Excluded vscode directory

Commit:
- Got provisioning running, no I need to handle ssh-keys

Commit:
- Provisioning with provided ssh-key works
- Started Containerization

Commit:
- Containerization works now

Commit:
- Containerization documented
- Id used now for stack as well, so one can provision multiple trainings

Commit:
- Finalized first prototype of AWS CDK JAva training infrastructure.

Commit:
- Outputs are now written to a json file containing the public dns and ips of the generated instances
- Enhanced the readme
- out directory is now used to populate the ssh-key-pairs and the generated outputs
- bash is now used to execute the script which gets installed now as well

ErhardSiegl commented:
Nachdem ich den Container nicht auf `ghcr.io/gepardec/train-aws-cdk` gefunden habe, wollte ich den Container selber bauen.

```
$ docker build -t train-aws-cdk -f Containerfile . 
...
 => ERROR [6/8] COPY cdk.context.json ./                                                                                                                                                                0.0s
------
 > [6/8] COPY cdk.context.json ./:
------
failed to compute cache key: "/cdk.context.json" not found: not found

erhards-gepard:aws-cdk esiegl$ grep -r  cdk.context.json .
./Containerfile:COPY cdk.context.json ./
./.gitignore:cdk.context.json
```
Was soll darin stehen und warum wird es kopiert, wo es im `.gitignore` steht?

cchet committed:
Update Containerfile

No cdk-context.json needed

cchet commented:
@ErhardSiegl  Die Datei wurde gelöscht, die wird beim command cdi synth/deploy erstellt. Ist bei mir nicht aufgefallen weil ich es auch lokal ausgeführt habe und da war die Datei schon vorhanden.

cchet commented:
@ErhardSiegl   Hattest du nochmal Zeit das auszuprobieren?

ErhardSiegl commented:
@cchet Habs ausprobiert und funktioniert. Für mein Training müsste ich noch die [Callback](https://github.com/Gepardec/JBossTrainings/blob/main/Schulungsumgebung/callbacks.sh) Funktion einbauen, die mir die Beschreibung und das Ansible Inventory generiert.

cchet commented:
@ErhardSiegl  Hast du es mit der bootstrap.sh probiert, die im /config directory sein muss. Die wird ausgeführt nachdem die VM gestartet wurde. Die installiert mir z.B beim Docker Training Docker auf Ubuntu.

ErhardSiegl commented:
Die bootstrap.sh funktioniert, aber die läuft auf den installierten Servern. Ich lass Ansible lokal laufen, damit ich leichter nachträglich was installieren kann. Daher brauche ich das inventory lokal. Ich habe es mir noch nicht genauer angesehen, bin aber optimistisch, dass es dafür eine Lösung gibt.

cchet committed:
- Added possibility of a callback.sh script
- Some minor renamings
- Deleted obsolete terraform resources
- Enhanced doc regarding callback.sh script
- Added cleanup of out directory on 'destroy' command
- modified github action to use new Containerfile

cchet committed:
- Deleted obsolete README.md and replaced it with new README.adoc

cchet commented:
@ErhardSiegl 
Das Feature Callbaclk habe ich implementiert.
Bitte README.adoc am Root vom Repository lesen.
Es sollte auch jemand die Doku anpassen und sich auch die github action anschauen, wo jetzt das AWS-CDK mit Java/Maven gebaut werden muss bevor man das Image bauen kann.

Auch nach einem Merge müssen die Training Repositories angepasst werden.

cchet commented:
@ErhardSiegl 
Ist dieser Pull-Request nicht mehr relevant?
